File: Create_And_Insertion_In_LinkedList.py
from types import new_class


class Node:

    def __init__(self, data) -> None:
        self.data = data
        self.next = None

class LinkedList:

    def __init__(self) -> None:
        self.head = None

    def push(self, data):
        newnode = Node(data)
        newnode.next = self.head
        self.head = newnode
    def insertAfter(self, prevnode, data):
        newnode = Node(data)
        if prevnode == None:
            print("Previous node cant be null")
            return
        else:
            newnode.next = prevnode.next
            prevnode.next = newnode
    def append(self, data):
        newnode = Node(data)
        last = self.head
        if self.head == None:
            self.head = newnode
            return
        
        while(last.next):
            last = last.next
        
        last.next = newnode
    
    def printList(self):
        temp = self.head
        while(temp):
            print(temp.data)
            temp = temp.next
        
def main():
    l1 = LinkedList()
    l1.push(7)
    l1.push(6)
    l1.push(5)
    l1.append(4)
    l1.push(2)
    l1.insertAfter(l1.head.next.next,3)
    l1.push(1)
    l1.printList()

if __name__ == "__main__":
    main()

File: Create_And_Insertion_In_LinkedList.cpp
#include<bits/stdc++.h>
using namespace std;

class Node{

	public:
	int data;
	Node *next;

};

void push(Node** head_ref, int data){

	Node* newnode = new Node();
	newnode->data = data;

	newnode->next = *head_ref;
	*head_ref = newnode;

}

void insertAfter(Node* prevNode, int data){
	if(prevNode->next==NULL){
		cout<<"The previous node cannot be null";
		return;
	}
	else{
		Node* newnode = new Node();
		newnode->data = data;
		newnode->next = prevNode->next;
		prevNode->next = newnode;

	}
}

void append(Node** head_ref, int data){

	Node* newnode = new Node();
	newnode->data = data;
	newnode->next = NULL;

	Node* last = *head_ref;

	if(*head_ref == NULL){
		*head_ref = newnode;
		return;
	}
	else{
		while(last->next!=NULL){
			last = last->next;
		}

	}
	
	last->next = newnode;
	return;

}

void printList(Node *node){
	while(node!=NULL){
		cout<<node->data<<" ";
		node = node->next;
	}

}

int main(){
	Node* head = NULL;
	append(&head, 6);
	push(&head, 5);
	push(&head, 7);
	push(&head, 4);
	push(&head, 2);
	insertAfter(head->next->next,1);
	push(&head, 9);
	printList(head);


	return 0;


}